### PR TITLE
PHOENIX-5780 Add mvn dependency:analyze to build process

### PR DESCRIPTION
**Pull request by stoty**
cleanup/reorganize dependency handling
also includes the following java code changes:
* remove accidental jline usages
* fix accidental shaded guava import
* replace commins-lang with commons-lang3 everywhere
* replace incidental commons-logging usages
* replace netty ThreadLocalRandom with j.u.c.ThreadLocalRandom
* replace com.sun.istack.NotNull with edu.umd.cs.findbugs.annotations.NonNull

### CODE REVIEW

File: phoenix-core/pom.xml [hunk 1]
```diff
@@ -212,30 +335,16 @@
...
       <groupId>org.antlr</groupId>
       <artifactId>antlr-runtime</artifactId>
     </dependency>
-    <dependency>
-      <groupId>jline</groupId>
-      <artifactId>jline</artifactId>
-    </dependency>
-    <dependency>
-      <groupId>sqlline</groupId>
-      <artifactId>sqlline</artifactId>
```
Comment (joshelser) [inline phoenix-core/pom.xml]: Dropping this here, removes sqlline from phoenix-client jar (via phoenix-core).

Either need to keep it here with an exclusion, or add sqlline as a dependency on phoenix-client.

File: phoenix-core/pom.xml [hunk 2]
```diff
@@ -167,8 +167,117 @@
   </build>
 
   <dependencies>
+    <!-- HBase compat dependency -->
+    <dependency>
+      <groupId>org.apache.phoenix</groupId>
+      <artifactId>phoenix-hbase-compat-${hbase.compat.version}</artifactId>
+    </dependency>
+    <!-- Haddop dependencies -->
```
[the remark this answers is not in the document]
Comment (stoty) [reply phoenix-core/pom.xml]: fixed

File: phoenix-pherf/pom.xml [hunk 3]
```diff
@@ -15,463 +15,546 @@
...
+        <version>${jaxb.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.glassfish.jaxb</groupId>
+        <artifactId>jaxb-runtime</artifactId>
+        <version>${jaxb.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>com.jcabi</groupId>
+        <artifactId>jcabi-jdbc</artifactId>
```
[the remark this answers is not in the document]
Comment (stoty) [reply phoenix-pherf/pom.xml]: Yes, there are some more of those, but I had to draw the line somewhere.
My favourite so far is having four places in the code where we process JSON, and using three different libraries for that.

File: phoenix-core/pom.xml (hunk 1 above)
Comment (stoty) [reply phoenix-core/pom.xml]: Right, I forgot because the code does not refer to it.
I feel that it belongs to client. I also updated it to 1.9 now.

**Comment by stoty** [comment]
> Trying to run a `mvn verify -DskipTests` locally (also failed in the same way with the 2.1 profile)

That's embarrassing. At least we know that the check works :) . Fixed.

**Pull request closed without merging**